corehq/apps/data_interfaces/utils.py

```diff
@@ -214,6 +215,23 @@ def _get_sql_repeat_record(domain, record_id):
         return None
 
 
+def _throttle_case_iterator(case_iterator):
+    """
+    Throttles slower queries that hog pgshard resources by sleeping for the amount of time the query took. Used
```

Comment:
I might reword this. We're throttling when a query is slow, but just the fact that a query is slow doesn't necessarily mean it's hogging pgshard resources—in the pattern motivating this, any individual query's time is mostly spent _waiting_, not spent hogging CPU cycles or something.

So it's more that we want this entire workload (case rule processing) to gently back off when query times are high.

Reply:
Thanks for clarifying. Updated in 6488628ba0a41a95a689a160da385adf6f2d706b.